crates/database/src/jit_orders.rs

```diff
@@ -0,0 +1,258 @@
+use {
+    crate::{
+        orders,
+        orders::{BuyTokenDestination, OrderKind, SellTokenSource, SigningScheme},
+        Address,
+        AppId,
+        OrderUid,
+    },
+    sqlx::{
+        types::{
+            chrono::{DateTime, Utc},
+            BigDecimal,
+        },
+        PgConnection,
+        QueryBuilder,
+    },
+};
+
+pub async fn single_full_jit_order(
+    ex: &mut PgConnection,
+    uid: &OrderUid,
+) -> Result<Option<FullJitOrder>, sqlx::Error> {
+    #[rustfmt::skip]
+        const QUERY: &str = const_format::concatcp!(
+"SELECT o.uid, o.owner, o.creation_timestamp, o.sell_token, o.buy_token, o.sell_amount, o.buy_amount,
+o.valid_to, o.app_data, o.fee_amount, o.kind, o.partially_fillable, o.signature,
+o.receiver, o.signing_scheme, o.sell_token_balance, o.buy_token_balance,
+(SELECT COALESCE(SUM(t.buy_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_buy,
+(SELECT COALESCE(SUM(t.sell_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_sell,
+(SELECT COALESCE(SUM(t.fee_amount), 0) FROM trades t WHERE t.order_uid = o.uid) AS sum_fee,
+COALESCE((SELECT SUM(surplus_fee) FROM order_execution oe WHERE oe.order_uid = o.uid), 0) as executed_surplus_fee",
+" FROM jit_orders o",
+" WHERE o.uid = $1 ",
+        );
+    sqlx::query_as(QUERY).bind(uid).fetch_optional(ex).await
+}
+
+/// Jit order combined with trades table and order_execution table, suitable for
+/// API responses.
+#[derive(Debug, Clone, Default, PartialEq, sqlx::FromRow)]
+pub struct FullJitOrder {
```

Comment:
It looks like this type is not needed. The API expects a `orders::FullOrder`. Instead of creating a new type (that has a very vague "Full" prefix which isn't really expressive) can we just explicitly return `Option<orders::FullOrder>` here (it's also not easy to find where the From trait is called and there is no strong reason to use this abstraction because the type is never used in a generic Vec or the like)

Reply:
If you want this then conversion from FullJitOrder to orders::FullOrder is either:
1. Part of sql query which is not reusable and not compiler checked
2. Part of `single_full_jit_order` function whish again in not reusable

Reusability is important since in a follow up PR I will implement similar function to `orders::full_orders_in_tx` that will reuse this.

I ended refactoring so that FullJitOrder is now private function and conversion is done inside jit_orders::single_full_jit_order which resolves most of your concerns.